src/web/twig/Extension.php

```diff
@@ -1017,12 +1026,45 @@ public function svgFunction($svg, bool $sanitize = null, bool $namespace = null,
             }
         }
 
-        if ($class !== null) {
-            $svg = preg_replace('/(<svg\b[^>]+\bclass=([\'"])[^\'"]+)(\\2)/i', "$1 {$class}$3", $svg, 1, $count);
+        // Load up the attributes
+        foreach ($attributes as $key => $value) {
```

Comment:
You should be able to build the attribute HTML using `Html::renderTagAttributes()`, just like the `html()` function uses.

Reply:
Sure, but you still need the loop - I think. We need to check whether any attributes already exist in the SVG and deal with them. I.E. If someone specifies a `class` key in the `$attr` array, but the SVG already has a class. 

The rendered result would need both the class that already existed and the class defined in `$attr`. So we can take the same approach as in the `$elements` loop. Build the HTML with an `HtmlHelper::**` function and then add if afterwards but not sure if that's what you mean?

Reply:
A rough sketch of how that function _might_ look: 
https://gist.github.com/gtettelaar/389e1ad76740193c803c69b8b725dcdb

Reply:
That doesn’t account for attribute names that are dynamically defined by `renderTagAttributes()`. For example if you pass `['data' => ['foo' => 'bar']]`, that will end up as `data-foo="bar"`.

Maybe best to just tack whatever attributes are defined here onto the end of the `<svg>` element, without changing any of the original attributes. If an element specifies the same attribute twice, browsers will just ignore the first one.

Reply:
> Maybe best to just tack whatever attributes are defined here onto the end of the <svg> element, without changing any of the original attributes. If an element specifies the same attribute twice, browsers will just ignore the first one.

What about [this](https://gist.github.com/gtettelaar/23fa9ba757889c60e78e8e37a029b114)?

Reply:
We can’t rely on the DOM extension for this, because that wasn’t built with SVGs in mind and errors out pretty easily.

I just added a new `|attr` filter that resolves #4660 (90c0eff97e3d4e55a53132e17217c2d40f92fd0b), so this change can be reverted completely now – except for the `$class` deprecation log.

Reply:
> I just added a new |attr filter that resolves #4660 (90c0eff), so this change can be reverted completely now – except for the $class deprecation log.

Nice! Reverted it

> We can’t rely on the DOM extension for this, because that wasn’t built with SVGs in mind and errors out pretty easily.

Just for my own curiosity do you have an article highlighting these flaws? I didn't come across this in any research, plus `enshrined/svg-sanitize` uses the DOM extension for SVG sanitation so Craft uses it indirectly even after this change was rolled back.

Reply:
To use it with SVGs you have to call `loadXML()` rather than `loadHTML`, but then it loses its ability to handle HTML syntax where it differs from strict XML rules (e.g. attributes that don’t have a value, and elements like `<br>` that aren’t self-closing but are void nonetheless). That’s fine for svg-sanitizer because SVGs are XML. But not reliable when embedding an SVG into an HTML document, at which point the SVG becomes HTML, and we need to be supporting HTML attribute syntax.

Reply:
Hmm, that's interesting! Thanks!